src/cascade/input_data/db/csmr.py

```diff
@@ -29,13 +30,36 @@ def _csmr_in_t3(execution_context):
     return exists == 1
 
 
+def _gbd_process_version_id_from_cod_version(cod_version):
+    query = """
+    SELECT gbd.gbd_process_version_metadata.gbd_process_version_id from gbd.gbd_process_version_metadata
+        JOIN  gbd.gbd_process_version ON gbd.gbd_process_version_metadata.gbd_process_version_id =
+                                         gbd.gbd_process_version.gbd_process_version_id
+        WHERE gbd_process_id = 3                     -- codcorrect process
+              and metadata_type_id = 1               -- codcorrect version
+              and val = %(cod_version)s
+              and gbd_process_version_status_id = 1  -- marked best
+    """
+
+    with cursor(database="gbd") as c:
+        c.execute(query, args={"cod_version": cod_version})
+        result = c.fetchone()
+
+    if result is None:
+        raise ValueError(f"No best gbd_process_version_id for cod version {cod_version}")
```

Comment:
If there weren't a best process version, this will make it fail... I don't know that it should recover. There is an InputDataError(Exception) for things the modeler can fix. Does this fall in that category? If not, it should have a category of things central comp wants to know.

Reply:
I new in advance that you would complain about this but even so I didn't do it right. Such is foresight. How about a GBDDataError?

Reply:
That sounds great.